### PR TITLE
Hide brightness slider on desktop experience

### PR DESCRIPTION
**Pull request by andirsun**
fixes: #163
We dont need to show the brightness slider if the user is on desktop or if have a laptop displaying only on external display. 

Added an smooth transition with a revealer.

### CODE REVIEW

File: src/Widgets/ScreenBrightness.vala [hunk 1]
```diff
@@ -54,9 +54,15 @@ public class Power.Widgets.ScreenBrightness : Gtk.EventBox {
...
         grid.add (brightness_slider);
 
-        add (grid);
+        var show_brightness_slider = new Gtk.Revealer ();
+        show_brightness_slider.add (grid);
 
-        brightness_slider.set_value (dm.brightness);
+        add (show_brightness_slider);
+
+        if (dm.brightness != -1) {
```
Comment (kvdz) [inline src/Widgets/ScreenBrightness.vala:62]: isn't there an other way to verify that the device doesn't have brightness control?  this seems a little hacky IMO

Comment (andirsun) [reply src/Widgets/ScreenBrightness.vala:62]: I didnt found other way. Maybe creating a property ?

**Comment by jeremypw** [comment]
Need to check whether the widget gets reconstructed if the display set up is changed otherwise `dm.brightness` will not be retested.

**Comment by andirsun** [comment]
@jeremypw Every time you display your laptop on docked mode the signal ``` dm.brightness_changed ``` is trigged.

File: src/Widgets/ScreenBrightness.vala (hunk 1 above)
Comment (kvdz) [reply src/Widgets/ScreenBrightness.vala:62]: That would be nice to see. Dm.brigthness.control_present (boolean) instead.

Comment (jeremypw) [reply src/Widgets/ScreenBrightness.vala:61]: Unless that property is going to be used in several places its probably not worth creating a new entity - I would just add an explanatory comment.

**Review by jeremypw — APPROVED** [review]
Works as expected thanks!

I think this should be merged and any tweaks to the UI can be done in other PRs rather than delaying any longer.

**Pull request merged**